server/server.go

```diff
@@ -144,12 +144,18 @@ func (svr *Server) Health(w http.ResponseWriter, r *http.Request) error {
 }
 
 func (svr *Server) HandleGet(w http.ResponseWriter, r *http.Request) error {
-
+	ctx := context.Background()
 	ct, err := ReadCommitmentMode(r)
 	if err != nil {
 		svr.WriteBadRequest(w, invalidCommitmentMode)
 		return err
 	}
+
+	actor := ReadActor(r)
```

Comment:
I assume we only want memstore to behave like a malicious actor. Can we not touch the code path from other storage type.

Reply:
we can but that'd increase the attack surface for critical system flows that likely shouldn't ever have modifiable return behaviors (i.e, comms with mainnet). we only should ever use this feature for testing sequencer behaviors when doing QA / experimentation

Reply:
So can we move actor inside memstore go files, so it no longer appears in the top level codeabse

Reply:
how would we do that when it needs to be read from the URL params?

Reply:
I guess my concern is that we only intend memstore to have the ability to behave maliciously, then we should localize the change to the memstore itsel. Because memstore is not intended for production use in any cases. Then I feel it is relatively safe. 

My understanding of of the code is that, it uses the context to determine proxy behavior. I worry it touches unnecesary part of code, making it bit harder to maintain. I worry we might accidentally inject some malicious behavior in the production path. Just my thoughts, I would invite @jianoaix to take a look.

Reply:
Is this server for production? Is production server taking the "actor" param from http request, based on which to decide the malicious behavior which is implemented in the store?

Reply:
it is but these malicious behavior patterns are only expressed within the memstore implementation which will never be used in production environments.